icepyx/core/query.py

```diff
@@ -1,445 +1,56 @@
-    Date range ['2019-02-20', '2019-02-28']
-
-    Initializing Query with a list of polygon vertex coordinate pairs.
-
-    >>> reg_a_poly = [(-55, 68), (-55, 71), (-48, 71), (-48, 68), (-55, 68)]
-    >>> reg_a_dates = ['2019-02-20','2019-02-28']
-    >>> reg_a = Query('ATL06', reg_a_poly, reg_a_dates)
-    >>> reg_a.spatial_extent
-    ('polygon', [-55.0, 68.0, -55.0, 71.0, -48.0, 71.0, -48.0, 68.0, -55.0, 68.0])
-
-    Initializing Query with a geospatial polygon file.
-
-    >>> from pathlib import Path
-    >>> aoi = Path('./doc/source/example_notebooks/supporting_files/simple_test_poly.gpkg').resolve()
-    >>> reg_a_dates = ['2019-02-22','2019-02-28']
-    >>> reg_a = Query('ATL06', str(aoi), reg_a_dates)
-    >>> print(reg_a)
-    Product ATL06 v006
-    ('polygon', [-55.0, 68.0, -55.0, 71.0, -48.0, 71.0, -48.0, 68.0, -55.0, 68.0])
-    Date range ['2019-02-22', '2019-02-28']
-
-    See Also
-    --------
-    GenQuery
-    """
 
+class Query(BaseQuery):
+    _temporal: Union[tp.Temporal, None]
     _CMRparams: apifmt.CMRParameters
-    _reqparams: apifmt.RequiredParameters
-    _subsetparams: Optional[apifmt.SubsetParameters]
-
-    # ----------------------------------------------------------------------
-    # Constructors
-
-    def __init__(
-        self,
-        product=None,
-        spatial_extent=None,
-        date_range=None,
-        start_time=None,
-        end_time=None,
-        version=None,
-        cycles=None,
-        tracks=None,
-        auth=None,
-        **kwargs,
-    ):
-        # Check necessary combination of input has been specified
-        if (product is None or spatial_extent is None) or (
-            (date_range is None and cycles is None and tracks is None)
-            and int(product[-2:]) <= 13
-        ):
-            raise ValueError(
-                "Please provide the required inputs. Use help([function]) to view the function's documentation"
-            )
+    REQUEST_RETRY_INTERVAL_SECONDS = 3
 
-        self._prod = is2ref._validate_product(product)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        super().__init__(spatial_extent, date_range, start_time, end_time, **kwargs)
+        # Ensure that the `_temporal` attr is set. This simplifies checking if
+        # temporal parameters has been passed. Instead of using `hasattr`, we
+        # can just check for `None`.
+        if not hasattr(self, "_temporal"):
+            self._temporal = None  # type: ignore[reportIncompatibleVariableOverride]
 
-        self._version = val.prod_version(is2ref.latest_version(self._prod), version)
+        logging.basicConfig(level=logging.WARNING)
 
-        # build list of available CMR parameters if reducing by cycle or RGT
-        # or a list of explicitly named files (full or partial names)
-        # DevGoal: add file name search to optional queries
+        cycles = kwargs.get("cycles")
+        tracks = kwargs.get("tracks")
```

Comment:
not sure where it's happening during execution, but I found the issue while reviewing the init so I'm noting it here: the search by cycle and tracks is not returning the correct results. It's returning far more granules than it should (and with cycles and tracks that are not specified as inputs).